homeassistant/components/namecheapdns/__init__.py

```diff
@@ -19,33 +20,52 @@
 
 UPDATE_URL = 'https://dynamicdns.park-your-domain.com/update'
 
+DEFAULT_HOST = '@'
+
 CONFIG_SCHEMA = vol.Schema({
     DOMAIN: vol.Schema({
         vol.Required(CONF_DOMAIN): cv.string,
         vol.Required(CONF_PASSWORD): cv.string,
-        vol.Optional(CONF_HOST, default='@'): cv.string,
-    })
+        vol.Optional(CONF_HOST, default=DEFAULT_HOST):
+        cv.string,
+        vol.Optional(CONF_HOSTS): cv.ensure_list,
+    }),
+    cv.deprecated(
+        CONF_HOST,
+        replacement_key=CONF_HOSTS,
+        invalidation_version='0.91'),
+    _ensure_list('hosts')
 }, extra=vol.ALLOW_EXTRA)
 
 
+def _ensure_list(key: str):
+    def validator(config: {}):
+        if key in config:
+            config[key] = cv.ensure_list(config[key])
+        return config
+    return validator
+
 async def async_setup(hass, config):
     """Initialize the namecheap DNS component."""
-    host = config[DOMAIN][CONF_HOST]
+    hosts = config[DOMAIN][CONF_HOSTS]
     domain = config[DOMAIN][CONF_DOMAIN]
     password = config[DOMAIN][CONF_PASSWORD]
 
     session = async_get_clientsession(hass)
 
-    result = await _update_namecheapdns(session, host, domain, password)
+    result = False
 
-    if not result:
-        return False
+    for host in hosts:
```

Comment:
I would recommend gathering all of these tasks into a list and then using `asyncio.wait` across the entire set at once. Similar example: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/media_player/squeezebox.py#L135

You can then iterate over the completed futures and check their values from there.